### ISSUE TITLE
Release protocol as crate

### ISSUE THREAD

**Issue by tychedelia** [Diggsey/franz#1]
@Diggsey, thank you for the wonderful work in this library, which is by far the most complete implementation of the Kafka protocol I've been able to find in Rust.

I've created a fork here https://github.com/0x1991babe/kafka-protocol making the following changes:

- Remove the client impl, focusing mainly on the protocol implementation.
- Updated all deps. The two commits required were (https://github.com/0x1991babe/kafka-protocol/commit/74aacf337a6ee75759932212b48f7e9a0bccbbd8, https://github.com/0x1991babe/kafka-protocol/commit/80781e612c9f4c096a913f56b5f08224ed061fae)
- Added AUTHORS file listing you as primary author and referenced your work in README.md.

If you have no objections, I'd like to add a license of your choosing and publish this as a crate for others to use.

**Comment by Diggsey** [comment]
I'm glad you have a use for it :)

That sounds great - my preferred license would be the standard MIT + Apache combo that's widely used for Rust crates.

**Comment by Diggsey** [comment]
The JSON files would need to retain their original license though as they come from Kafka.

**Comment by tychedelia** [comment]
Added the dual license. I removed the JSON schemas in favor of dynamically downloading + generating (https://github.com/0x1991babe/kafka-protocol-rs/commit/1b1a132beb767a3613e12136a243bb7ae1401e67), and fixed some issues with recent updates to the protocol as well. I think this is proof of this project's meta-programming approach -- they've like doubled the API size since you wrote this just a few years ago. 

I'm going to add CI and a few more tests before publishing, will close this issue to let you know then. Thanks so much again!
